packages/data-transport-layer/src/services/l1-ingestion/service.ts

```diff
@@ -402,20 +407,23 @@ export class L1IngestionService extends BaseService<L1IngestionServiceOptions> {
   }
 
   /**
-   * Gets the address of a contract at a particular block in the past.
+   * Gets the address of a contract at a particular block in the past through
+   * searching between a block range for AddressSet events.
    *
    * @param contractName Name of the contract to get an address for.
-   * @param blockNumber Block at which to get an address.
+   * @param fromL1Block Block at which to start the search
+   * @param toL1Block Block at which to stop the search
    * @return Contract address.
    */
-  private async _getContractAddressAtBlock(
+  private async _findLatestNewAddressBetweenBlockRange(
     contractName: string,
-    blockNumber: number
+    fromL1Block: number,
+    toL1Block: number
   ): Promise<string> {
     const events = await this.state.contracts.Lib_AddressManager.queryFilter(
       this.state.contracts.Lib_AddressManager.filters.AddressSet(contractName),
-      this.state.startingL1BlockNumber,
-      blockNumber
+      fromL1Block,
```

Comment:
I believe this is broken in the case that the last `AddressSet` event happened *before* the `fromL1Block`. I think we need to brainstorm on a better way to make this more efficient, maybe we can store some values in the `Lib_AddressManager` that allow us to do a binary search or something.